File: src/Frontend/projects/core/src/lib/models/dto/pim/attribute-value.ts
export class AttributeValue {
    public id: number;
    public listValueId: number;
    public strValue: string;
    public numValue?: number;
    public boolValue?: boolean;
    public attributeId: number;
    public productId: number;
    public creatorId: number;
    public dateValue?: Date;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }
        return Object.assign(
            new AttributeValue(),
            obj,
            {
                dateValue: obj.dateValue ? new Date(obj.dateValue) : null,
            },
        );
    }
}

File: src/Backend/Common/Models/Pim/AttributeValue.cs
using System;
using NpgsqlTypes;

namespace Company.Common.Models.Pim
{
    public class AttributeValue
    {
        public int Id { get; set; }

        public int? ListValueId { get; set; }

        public ListValue ListValue { get; set; }

        public string StrValue { get; set; }

        public string SearchString { get; set; }

        public double? NumValue { get; set; }

        public bool? BoolValue { get; set; }

        public DateTime? DateValue { get; set; }

        public DateTime CreateTime { get; set; }

        public int? CreatorId { get; set; }

        public int AttributeId { get; set; }

        public Attribute Attribute { get; set; }

        public int ProductId { get; set; }

        public Product Product { get; set; }

        public NpgsqlTsVector SearchVector { get; set; }

    }
}
